File: python/0415_add_strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        i, j = len(num1) - 1, len(num2) - 1
        carry = 0
        res_str = ''
        while i >= 0 or j >= 0:
            sum_int = 0
            if i >= 0:
                sum_int += int(num1[i])
            if j >= 0:
                sum_int += int(num2[j])
            sum_int += carry
            if sum_int >= 10:
                carry = 1
                sum_int = sum_int - 10
            else:
                carry = 0
            res_str += str(sum_int)
            i -= 1
            j -= 1
        if carry:
            res_str += '1'

        return res_str[::-1]


print(Solution().addStrings('1999', '1'))
print(Solution().addStrings('2319082', '12314'))

File: cpp/0415_add_strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
      if (num1.length() > num2.length()) {
        return addStrings(num2, num1);
      }
      int len1 = num1.length();
      int len2 = num2.length();

      bool flag = false;
      int temp = 0;
      for(int i = 0; i < len2; i++) {
        if (i < len1) {
          temp = num1[len1-1-i] + num2[len2-1-i] - '0'*2 + flag;
        } else {
          temp = num2[len2-1-i] - '0' + flag;
        }
        flag = temp > 9;
        num2[len2-1-i] = temp % 10 + '0';
      }
      if (flag) {
        return '1'+num2;
      } else {
        return num2;
      }
    }
};
